crypto.go

```diff
@@ -10,29 +10,41 @@ import (
 	"golang.org/x/crypto/nacl/secretbox"
 )
 
-func keyedHash(key, text []byte) []byte {
+func keyedHash(key *[hashLen]byte, text []byte) *[hashLen]byte {
 	h := keyedHasher(key)
 	h.Write([]byte(text))
-	return h.Sum(nil)
+	var sum [hashLen]byte
+	r := h.Sum(sum[:0])
+	if &r[0] != &sum[0] {
```

Comment:
The Sum function is defined to append the result to its argument. Append in Go will use existing space in the slice if it can - otherwise it will return a new slice. In this case we are sure that the result fits into the slice,  but just to make sure we check the address of the first element which will be different if a new slice has been allocated. This will work with any Go compiler. It does not help if the result  is shorter than expected,  but that case doesn't matter so much as we have at least put the result of the hash into the right place.